pkg/anago/release.go

```diff
@@ -91,6 +97,31 @@ func (d *defaultReleaseImpl) PrepareWorkspaceRelease(
 	return release.PrepareWorkspaceRelease(directory, buildVersion, bucket)
 }
 
+func (d *defaultReleaseImpl) CreateAnnouncement(
+	directory, buildVersion, releaseBranch string,
+) error {
+	// Get a semver from the buildVersion
+	semver, err := util.TagStringToSemver(buildVersion)
+	if err != nil {
+		return errors.Wrap(err, "parsing version tag into semver")
+	}
+
+	// Buld the announcement options set
+	announceOpts := announce.NewOptions()
+	announceOpts.WithWorkDir(directory)
+	announceOpts.WithTag(util.SemverToTagString(semver))
+	announceOpts.WithBranch(releaseBranch)
+	announceOpts.WithChangelogPath(
+		filepath.Join(gitRoot, fmt.Sprintf("/CHANGELOG/CHANGELOG-%d.%d.md", semver.Major, semver.Minor)),
+	)
+	announceOpts.WithChangelogHTML(
+		filepath.Join(workspaceDir, "/src/release-notes.html"),
+	)
+
+	// Create the announcement
+	return announce.CreateForRelease(announceOpts)
+}
```

Comment:
Let's only call single functions in those mock implementations, for example `announce.CreateForRelease`. The question is when can this call fail? I would say only on `announce.CreateForRelease`. We do not care about the internal implementation of this function, that's why we mock it away.

Reply:
OK, I've moved the logic for building the logic to the client and out of the implementation.